.github/workflows/pre-merge.yaml

```diff
@@ -22,15 +22,12 @@ jobs:
       - name: Checkout Repo
         uses: actions/checkout@v2
       - name: Cache Gradle Caches
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
-          path: ~/.gradle/caches/
-          key: cache-gradle-cache
-      - name: Cache Gradle Wrapper
-        uses: actions/cache@v1
-        with:
-          path: ~/.gradle/wrapper/
-          key: cache-gradle-wrapper
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/**.gradle*', '**/gradle/wrapper/gradle-wrapper.properties', '**/libs.versions.toml') }}
```

Comment:
Let's merge this as it is, but we probably want to go with https://github.com/gradle/gradle-build-action

Would you be up for a follow-up?

Reply:
I'll take a look.